### PR TITLE
feat: Release notes for Codacy Cloud October 2023 DOCS-446

### PR DESCRIPTION
**Pull request by nicklem**
Adds auto-generated release notes for Codacy Cloud October 2023

### :eyes: Live preview
https://feat-release-notes-cloud-2023-10--docs-codacy.netlify.app/release-notes/cloud/cloud-2023-10/

### :construction: To do
- [x] Add the new release notes page to `mkdocs.yml` and `docs/release-notes/index.md`
- [x] Review generated release notes
- [x] Review list of issues with missing release notes
- [x] Review links to changelogs of updated tools

Additional To do
- [x] Manually add https://codacy.atlassian.net/browse/ALA-593
- [x] Manually add https://codacy.atlassian.net/browse/IO-830

### CODE REVIEW

File: docs/release-notes/cloud/cloud-2023-10.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+📢 [Visit the Codacy roadmap](https://roadmap.codacy.com) and <span class="skip-vale">let us know</span> your feedback on both new and planned product updates!
+
+## Product enhancements
+
+-   You can now get two additional coverage status reports on GitHub using the new faster Coverage engine, showing whether your diff coverage and coverage variation are up to standards or not as configured on the quality gate rules for your repository. See [how to activate this setting](../../repositories-configure/integrations/github-integration.md#new-coverage-engine) for a repository. (ALA-593)
+
+    ![New Coverage status report](../images/ala-593.png)
+
+-   Codacy will drop support for branches named HEAD or matching the pattern refs/heads/*. These names are reserved for git and shouldn't be used as branch names. (IO-830)
```
Comment (nicklem) [inline docs/release-notes/cloud/cloud-2023-10.md:22]: @claudiacarpinteiro Can you review this one? 🙏

File: docs/release-notes/cloud/cloud-2023-10.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+## Bug fixes
+
+-   Fixed an issue that caused code duplication warnings to link to the original code block instead of the duplicated one. (TS-545)
+-   TSQLLint can now also detect and analyze files with an `sql` extension. (TS-543)
+-   Analysis retries now follow the same permissions configured in Analysis Configuration (configurable in Organization settings / Member privileges / Analysis configuration) allowing a members to actually ignore issues since they can now reanalyse commits and pull requests (PLUTO-587)
```
Comment (nicklem) [inline docs/release-notes/cloud/cloud-2023-10.md:28]: @claudiacarpinteiro Can you review this one? 🙏

File: docs/release-notes/cloud/cloud-2023-10.md (hunk 1 above)
Comment (claudiacarpinteiro) [reply docs/release-notes/cloud/cloud-2023-10.md:22]:
```suggestion
-   Codacy no longer supports and now skips the analysis of branches named `HEAD` or matching the pattern `refs/heads/*`, as these are Git reserved terms. (IO-830)
```

Comment (claudiacarpinteiro) [reply docs/release-notes/cloud/cloud-2023-10.md:22]: I'm, not sure if we want to announce this as a product enhancement...

@MiguelMarcelino WDYT?
Does this change fit a product enhancement, or should we announce it as a bug fix? 🤔

File: docs/release-notes/cloud/cloud-2023-10.md (hunk 2 above)
Comment (claudiacarpinteiro) [reply docs/release-notes/cloud/cloud-2023-10.md:28]: @nicklem I already clarified with @lolgab, this one is a **product enhancement** (although it started as a bug fix). 

Here's my review:
```suggestion
-   The permission level set for the **Analysis configuration** on the organization **Settings > Member privileges** now applies also to the reanalysis of a pull request or the last commit in a branch. (PLUTO-587)
```

File: docs/release-notes/cloud/cloud-2023-10.md (hunk 1 above)
Comment (MiguelMarcelino) [reply docs/release-notes/cloud/cloud-2023-10.md:23]: Not entirely sure. For me, this looks like a bug. However, I would say that calling it a bug vs. enhancement depends on what we consider to be the source of truth. If the provider is the source of truth, then it is an enhancement. Despite our supported providers allowing us to create these branches, some providers don't behave that well when this is done. If we want to use git as the source of truth, then this is a bug.
But I would maybe ask someone from @codacy/io to give their opinion about this as well.

Comment (pedrocodacy) [reply docs/release-notes/cloud/cloud-2023-10.md:23]: I would tend to agree with @MiguelMarcelino's initial assessment, this is a bug imo

**Review by pedrocodacy — APPROVED** [review]
from Io's side looks good 🤘

**Pull request merged**